### ISSUE TITLE
deepseek_v4.model_npu_fused 报错

### ISSUE THREAD

**Issue by singzhou** [singzhou/nano-deepseek-v4-inference#1]
root@master1:/opt/z00830407/nano-deepseek-v4-inference# python -m deepseek_v4.model_npu_fused --device npu
INFO 09-01 14:07:37 [__init__.py:52] Available plugins for group vllm.platform_plugins:
INFO 09-01 14:07:37 [__init__.py:54] - ascend -> vllm_ascend:register
INFO 09-01 14:07:37 [__init__.py:57] All plugins in this group will be loaded. Set `VLLM_PLUGINS` to control which plugins to load.
INFO 09-01 14:07:37 [__init__.py:237] Platform plugin ascend is activated
INFO 09-01 14:07:37 [platform.py:70] Breakable CUDAGraph on Ascend is opt-in; using VLLM_USE_BREAKABLE_CUDAGRAPH=0.
[W901 14:07:43.114667950 TensorFactories.cpp:340] Warning: Cannot create tensor with interal format while allow_internel_format=False, tensor will be created with base format. (function operator())
[W901 14:07:44.530196050 ArgSortKernelNpuOpApi.cpp:26] Warning: Warning: kernel [ArgSort] can not support dtype int32 or int64 on AiCore, Now this kernel is running on AiCpu.If you are more concerned about high-performance execution,please cast dtype to float32. (function operator())
device: npu:0, dtype: torch.bfloat16
torch_npu available: True
vLLM-Ascend custom ops loaded: True
prompt shape: (1, 9)
generated shape: (1, 11)
generated ids: [[10, 20, 30, 40, 50, 60, 70, 80, 90, 26, 492]]

KV Cache:
  {'next_position': 11, 'local_tokens': 8, 'compressed_tokens': 0, 'compress_ratio': 0, 'layer': 0, 'attention_type': 'SWA'}
  {'next_position': 11, 'local_tokens': 8, 'compressed_tokens': 2, 'compress_ratio': 4, 'layer': 1, 'attention_type': 'CSA'}
  {'next_position': 11, 'local_tokens': 8, 'compressed_tokens': 1, 'compress_ratio': 8, 'layer': 2, 'attention_type': 'HCA'}

融合路径（逻辑算子 -> 实际 backend）:
  mHC-pre          -> torch_fallback（NPU kernel 拒绝当前 shape: hc_mult only supports 4, actual 2.）
    IO: x(1, 1, 2, 64) -> reduced(1, 1, 64), post(1, 1, 2), comb(1, 1, 2, 2)
  RMSNorm          -> torch_npu.npu_rms_norm
    IO: x(1, 1, 64) + weight(64,) -> out(1, 1, 64)
  PartialRoPE      -> _C_ascend.inplace_partial_rotary_mul
    IO: x(1, 4, 16) + cos/sin[4] -> out(1, 4, 16)
  Grouped-OProj    -> torch_fallback（NPU kernel 拒绝当前 shape: npu_transpose_batchmatmul:../third_party/op-plugin/op_plugin/ops/opapi/TransposeBatchMatMulKernelNpuOpApi.cpp:67 NPU function error: call aclnnTransposeBatchMatMul failed, error code is 161002
[ERROR] 2026-09-01-14:07:44 (PID:326850, Device:0, RankID:-1) ERR00100 PTA call acl api failed.
[PID: 326850] 2026-09-01-14:07:44.627.361 AclNN_Parameter_Error(EZ1001): The k-axis of the two inputs are different [1, 2, 32], [2, 16, 32]
）
    IO: x(1, 2, 32) @ weight(2, 16, 32) -> out(1, 2, 16)
  mHC-post         -> torch_fallback（NPU kernel 拒绝当前 shape: Input tensor x's dim num should be 3, actual 4.）
    IO: branch(1, 1, 64) + residual(1, 1, 2, 64) + post(1, 1, 2) + comb(1, 1, 2, 2) -> out(1, 1, 2, 64)
  MoE-router       -> _C_ascend.moe_gating_top_k_hash
    IO: logits(1, 4) -> weights(1, 2) + ids(1, 2)
  SwiGLU           -> torch_npu.npu_swiglu
    IO: merged_gate_up(1, 256) -> out(1, 128)
